core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/CompactEntityComponent.scala

```diff
@@ -731,170 +653,98 @@ class CompactEntityQuery(driverComponent: DriverComponent)
           and name = $oldName
           and deleted = 0"""
 
-    // Update the entity from_name in the ENTITY_REFS table
-    // explain plan: index range scan on unq_from_to
-    val updateFromNameSql =
-      sql"""update ENTITY_REFS
-            set from_name = $newName
-            where workspace_id = $workspaceId
-            and from_entity_type = $entityType
-            and from_name = $oldName"""
-
-    // Update the entity to_name in the ENTITY_REFS table
-    // explain plan: index range scan on unq_from_to
-    val updateToNameSql =
-      sql"""update ENTITY_REFS
-            set to_name = $newName
+    // Update all embedded references in the $.refs array.
+    // This is done via JSON_REPLACE(CAST(REPLACE(JSON_EXTRACT))). Explaining from the inside out:
+    //   - JSON_EXTRACT(attributes, '$$.refs') gets the refs array
+    //   - REPLACE(...) treats the refs array as a plain string, and replaces all occurrences of
+    //      the old name with the new name
+    //   - CAST(... as JSON) converts the modified string back to a JSON array
+    //   - JSON_REPLACE(...) replaces the original refs array with the modified one
+    // We use a string replace here because it is significantly more performant than calling JSON_REPLACE
+    //  individually for each value that needs to be changed. Since we control the serialization format of the
+    //  refs array, and only perform the string replace inside that array, we avoid any problems with other
+    //  user-supplied values.
+    val updateReferencesInAttributesSql = sql"""update ENTITY
+            set attributes = JSON_REPLACE(attributes, $slickRefsPath,
+              CAST(REPLACE(JSON_EXTRACT(attributes, $slickRefsPath), '"n": "#$oldName", "t": "#$entityType"', '"n": "#$newName", "t": "#$entityType"') as JSON))
```

Comment:
Is there a reason to use `#$`? I realize `oldName`, `entityType`, and `newName` are probably validated and yet I still worry about injection.

Would something like this work:
```
val oldRef = s""""n": "$oldName", "t": "$entityType""""
val newRef = s""""n": "$newName", "t": "$entityType""""
CAST(REPLACE(JSON_EXTRACT(attributes, $slickRefsPath), $oldRef, $newRef) as JSON))
```
also more readable, I think.

Reply:
I like this approach much better - adopted in [b9276db](https://github.com/broadinstitute/rawls/pull/3338/commits/b9276db498f39c2f6f59f4313886310d065c7474), thank you!